src/heal_location.c

```diff
@@ -35,3 +38,63 @@ const struct HealLocation *GetHealLocation(u32 index)
     else
         return &sHealLocations[index - 1];
 }
+
+bool32 IsLastHealLocation(u32 healLocation)
+{
+    const struct HealLocation *loc = GetHealLocation(healLocation);
+    const struct WarpData *warpData = &gSaveBlock1Ptr->lastHealLocation;
+
+    return warpData->mapGroup == loc->group
+        && warpData->mapNum == loc->map
+        && warpData->warpId == WARP_ID_NONE
+        && warpData->x == loc->x
+        && warpData->y == loc->y;
+}
+
+static void SetWhiteoutRespawnHealerNpcAsLastTalked(u32 healLocationId)
+{
+    gSpecialVar_LastTalked = sWhiteoutRespawnHealerNpcLocalIds[healLocationId - 1];
+}
+
+void SetWhiteoutRespawnWarpAndHealerNpc(struct WarpData * warp)
+{
+        u32 healLocationId = GetHealLocationIndexByMap(gSaveBlock1Ptr->lastHealLocation.mapGroup, gSaveBlock1Ptr->lastHealLocation.mapNum);
+        warp->mapGroup = sHealLocationsPokemonCenter[healLocationId - 1][0];
+        warp->mapNum = sHealLocationsPokemonCenter[healLocationId - 1][1];
+        warp->warpId = 0xFF;
+
+        if (sHealLocationsPokemonCenter[healLocationId - 1][0] == MAP_GROUP(LITTLEROOT_TOWN_BRENDANS_HOUSE_1F) && sHealLocationsPokemonCenter[healLocationId - 1][1] == MAP_NUM(LITTLEROOT_TOWN_BRENDANS_HOUSE_1F))
```

Comment:
Since we can never both have FRLG whiteouts and the normal ones, I wonder if these hardcoded coordinates should instead become part of a config-dependent `sHealLocations`. Just an idea

Reply:
`HEAL_LOCATION_SOUTHERN_ISLAND_EXTERIOR` uses the normal whiteout even with FRLG enabled. I think it might also be possible to replace `struct WarpData lastHealLocation` in `SaveBlock1` with just a `u8 healLocationId` saving 4 byte. This would also make checking for the location much easier. Though this would obviously be save breaking.

Reply:
Does HEAL_LOCATION_SOUTHERN_ISLAND_EXTERIOR exist in ORAS? What does it do there?

Reply:
Not sure, but it is only used as a flight destination, but is never set, since it is not possible to target Southern Island as a destination. I don't think it was ever intended as a heal location, just planned as a possible flight path. HEAL_LOCATION is imo a bit of a misleading name, since Teleport and Fly also use them.